common/repo/GitRepo.java

```diff
@@ -366,33 +299,65 @@ public void removeBranch(String branch) {
             .collect(Collectors.toList()));
   }
 
+  @Override
   public String getFileContents(String commitId, String path) {
-    try (ObjectReader objectReader = jGitRepo.newObjectReader();
-        RevWalk revWalk = new RevWalk(objectReader)) {
-      final ObjectId objectId = jGitRepo.resolve(commitId);
-      RevCommit revCommit = revWalk.parseCommit(objectId);
-      RevTree revTree = revCommit.getTree();
-      byte[] data;
-      try (TreeWalk treeWalk = TreeWalk.forPath(objectReader, path, revTree)) {
-        if (treeWalk == null) {
-          throw new IllegalStateException(
-              String.format("TreeWalk is null for commitId %s and path %s", commitId, path));
-        }
-        // Index 0 should have file data
-        data = objectReader.open(treeWalk.getObjectId(0)).getBytes();
-      }
-      return new String(data, UTF_8);
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    CommandResult commandResult = runCommand("show " + commitId + ":" + path);
+    String result = commandResult.stdout;
+    int lastIndexOfNewLineSymbol = result.lastIndexOf("\n");
```

Comment:
Can you explain this code? (lines 306-311)

Reply:
Actually it needed for tests. I some changed tests and deleted this code.